### PR TITLE
Create empty Job Service

### PR DESCRIPTION
**Pull request by tsotnet**
**What this PR does / why we need it**: This PR creates an empty Job Service with and the Docker image creation step.

### CODE REVIEW

File: infra/docker/jobservice/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.7-slim-buster
+
+USER root
+WORKDIR /feast
+
+COPY sdk/python sdk/python
+COPY Makefile Makefile
+COPY protos protos
+COPY tests tests
```
Comment (woop) [inline infra/docker/jobservice/Dockerfile:9]: Do we need the tests?

File: protos/feast/core/JobService.proto [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import "feast/serving/ServingService.proto";
+
+service JobService {
+    // Start job to ingest data from offline store into online store
+    rpc StartOfflineToOnlineIngestionJob (StartOfflineToOnlineIngestionJobRequest) returns (StartOfflineToOnlineIngestionJobResponse);
+
+    // Produce a training dataset, return a job id that will provide a file reference
+    rpc GetHistoricalFeatures (GetHistoricalFeaturesRequest) returns (GetHistoricalFeaturesResponse);
+
+    //   IMPORTANT: The following RPCs are not must-haves for the MVP. They are stretch goals. All of these methods would exist internally in the job service and would be called by a control loop to start/stop/restart streaming ingestion jobs.
```
Comment (woop) [inline protos/feast/core/JobService.proto:35]: We can probably just remove this.

File: protos/feast/core/JobService.proto [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    // The Historical Features request that triggered this export job
+    GetHistoricalFeaturesRequest request = 2;
+  }
+
+  message OfflineToOnlineMeta {
+    // Reference to the Feature Table being populated by this job
+    string project = 1;
+    string table_name = 2;
+    // Connection String to the Redis Online Store that the job is writing to.
+    string redis_connection_string = 3;
```
Comment (oavdeev) [inline protos/feast/core/JobService.proto:97]: we should make this a struct for consistency with the[ ingestion job parameter format](https://github.com/feast-dev/feast/blob/master/spark/ingestion/src/main/scala/feast/ingestion/IngestionJobConfig.scala#L29)

**Review by terryyylim — COMMENTED** [review]
I think we should standardize the naming to `created_timestamp`, `stop_timestamp` since those are use eg in FeatureTable proto as well.

File: protos/feast/core/JobService.proto [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  // Batch DataSource that can be used to obtain entity values for historical retrieval.
+  // For each entity value, a feature value will be retrieved for that value/timestamp
+  // Only 'BATCH_*' source types are supported.
+  // Currently only BATCH_FILE source type is supported.
+  DataSource entities_source = 2;
+
+  // Optional field to specify project name override. If specified, uses the
+  // given project for retrieval. Overrides the projects specified in
+  // Feature References if both are specified.
+  string project_name = 3;
```
Comment (terryyylim) [inline protos/feast/core/JobService.proto:147]: Maybe standardize to `project` here?

File: sdk/python/feast/job_service.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+from concurrent.futures import ThreadPoolExecutor
+
+import grpc
+
+import feast
+from feast.core import JobService_pb2_grpc
+
+
+class JobServiceServicer(JobService_pb2_grpc.JobServiceServicer):
```
Comment (terryyylim) [inline sdk/python/feast/job_service.py:9]: I recommend we follow pep8 style guide as stated [here](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names), to utilize lowercase separated by underscores for function names.

**Comment by tsotnet** [comment]
I just added the helm charts, I'll be working on handling the above comments

File: infra/charts/feast/charts/feast-jobservice/templates/deployment.yaml [hunk 6]
```diff
@@ -109,22 +109,21 @@ spec:
...
+        #   {{- end }}
+        #   {{- if index .Values "application-generated.yaml" "enabled" -}}
+        #   ,file:/etc/feast/application-generated.yaml
+        #   {{- end }}
+        #   {{- if index .Values "application-secret.yaml" "enabled" -}}
+        #   ,file:/etc/secrets/feast/application-secret.yaml
+        #   {{- end }}
+        #   {{- if index .Values "application-override.yaml" "enabled" -}}
+        #   ,file:/etc/feast/application-override.yaml
+        #   {{- end }}
```
Comment (oavdeev) [inline infra/charts/feast/charts/feast-jobservice/templates/deployment.yaml:126]: remove commented out stuff?

Comment (tsotnet) [reply infra/charts/feast/charts/feast-jobservice/templates/deployment.yaml]: Will do in a separate diff

File: sdk/python/feast/job_service.py (hunk 5 above)
Comment (tsotnet) [reply sdk/python/feast/job_service.py:9]: This is autogenerated code by grpc, I don't think we have a control to modify it.

File: protos/feast/core/JobService.proto (hunk 3 above)
Comment (oavdeev-tt) [reply protos/feast/core/JobService.proto:97]: Actually lets just drop it from here for now. The only way this entire `Job` proto is used is to list jobs, i think it is fine if you just get the basic info like job type and table name there, but not redis connection details. 

We can always add more stuff later.

File: protos/feast/core/JobService.proto [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    // Get details of a single job
+    rpc GetJob (GetJobRequest) returns (GetJobResponse);
+}
+
+
+enum JobType {
+	INVALID_JOB = 0;
+	OFFLINE_TO_ONLINE_JOB = 1;
+	STREAM_TO_ONLINE_JOB = 2;
+	EXPORT_JOB = 4;
```
Comment (khorshuheng) [inline protos/feast/core/JobService.proto:53]: Historical feature retrieval job would be classify as an export job?

File: infra/docker/jobservice/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+USER root
+WORKDIR /feast
+
+COPY sdk/python sdk/python
+COPY Makefile Makefile
+COPY protos protos
+
+# Install make
+RUN apt-get update && apt-get -y install make git
```
Comment (khorshuheng) [inline infra/docker/jobservice/Dockerfile:11]: Why do we need git for the dockerfile?

File: infra/docker/jobservice/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+# Install Python dependencies
+RUN make compile-protos-python
+
+# Install Feast SDK
+COPY .git .git
```
Comment (khorshuheng) [inline infra/docker/jobservice/Dockerfile:17]: Is there a reason why we copy git into the docker file?

**Review by woop — APPROVED** [review]

File: infra/docker/jobservice/Dockerfile (hunk 9 above)
Comment (tsotnet) [reply infra/docker/jobservice/Dockerfile:17]: Looks like git is used by sdk/python/setup.py here https://github.com/feast-dev/feast/blob/master/sdk/python/setup.py#L52. If I remove this line docker build step errors out.

Also I copied this mostly from the jupyter dockerfile, which contains the similar build step. We could improve docker build files by removing the dependency on git (seems like it's only used to copy the readme file from the git repo root), but I think that's probably better handled by another PR.

**Pull request merged**